app/models/post.rb

```diff
@@ -1,2 +1,13 @@
 class Post < ApplicationRecord
+  belongs_to :user
+  has_many :like
+  has_many :comment
+
+  def update_counter
+    user.increment(:post_counter)
+  end
```

Comment:
- [ ] You should use the `author ` instead of the `user` attribute to call the `increment` method
- [OPTIONAL] You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the ` #update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the `update_post_counter` method private

```ruby
 after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```